### ISSUE TITLE
Slow rendering on first change

### ISSUE THREAD

**Issue by rjahn** [sibvisions/reactUI#463]
Open a screen with many fields and no records.

Insert a new record -> Detail will show a white area before all editors will be shown.
No problems after first time!

https://github.com/sibvisions/reactUI/assets/1307043/b8a1d072-8ab4-45a9-93ba-8684b7a1e0d2

**Comment by rjahn** [comment]
performance is much better with latest primereact and master branch, but still a bit flickering.

**Comment by gimmixAT** [comment]
@rjahn could you point me to an example screen or project where this is reproducible in case this is still an issue that needs to be fixed within this milestone

**Comment by rjahn** [comment]
Demo screen is available (User: bug, Screen: Slow rendering)

The screen doesn't persist records - it's just in memory.
The problem is reproducable, after every open/re-open. The first insert lets the detail flickr.

Please, could you check why the fields: Password, Html Text are NOT gray? and the password field shows the focus rect as well.

<img width="1236" alt="image" src="https://github.com/user-attachments/assets/7132860e-bf57-4aaf-afe0-8b2f4cfc3d1d">

In Swing, all fields are disabled: 

<img width="698" alt="image" src="https://github.com/user-attachments/assets/405e1426-8ec1-49f9-89d2-f093a6b65e3f">

It's also possible that this is a server problem, but not sure right now.

**Comment by gimmixAT** [comment]
The styling of the disabled date & HTML Field should be as expeected now

The reason why there is such a noticable delay / white screen, between adding the first new item and additional items, is that the first screen (with all disabled fields) is actually completely discarded via a server command and a new one is sent. Afterward when adding new items the screen is kept around and just updated.

On top of that the screen has a lot of date picker fields which seems to be quite an expensive component due to primereact chose to implement it (precalcs some things on render instead of on demand).

**Comment by rjahn** [comment]
This sounds strange, because it's the same screen. If you click on "+ Neu" only enabled state should be changed. The server shouldn't send new components.

**Comment by gimmixAT** [comment]
with the pressButton request I get a list of changedComponents that contains a single remove entry which seems to be the parent panel -> `SloRen-FS_MP1-Intern-SCROLLP_DETAIL`

**Comment by rjahn** [comment]
You're absolutely right!
We could optimize the server part, but currently it's a good use-case for another problem...

I guess that there are more UI cycles on client?
The "layout handler" should remove the panel, add the new one, re-layout and afterwards the browser should show the changes (all in same UI cycle). Looks like there are different repaints for: remove (UI cycle 1), add + re-layout (UI cycle 2).

Also, sub panel didn't change layout relevant information, so re-calculation shouldn't be necessary... but this depends on client layout implementation.

**Comment by rjahn** [comment]
If a developer creates a new panel on button click and replaces an old panel with the new panel, it also would flickr because of client repaints? This shouldn't be the case and we can't optimize the server-side in this case.

The current issue is only a show-case but not for slow rendering of react.

**Comment by gimmixAT** [comment]
The problem as far as react and our current render stack therin is concerned here is that we we use the component id (e.g. `P123`) as a `key`. When that changes React won't reuse a component even if the content would technically be the same. Which in this case causes a complete rerender.
We also can't just simply remove that because the render stack relies on that key to change to properly rerender the currently displayed UI after changes sent by the server.

**Comment by rjahn** [comment]
In our case, the key (id) should be the same for the component because we remove it on server-side and simply add it again as sub component of a new scroll panel.

You mean that the new scroll panel forces a re-render?

**Comment by gimmixAT** [comment]
Yes because the new scroll panel has a different id the key changes and the component and all its children are discarded instead of being reused. I agree ideally this shouldn't be the case but changing that will probably need a larger refactor of the current render stack. 
I already attempted to get rid of the key but go stuck at an infinite loop in some layout algorithm.

**Comment by rjahn** [comment]
We will move this issue to a later version. This takes more time to implement and is currently not a big problem.
